library/Icinga/Util/Csp.php

```diff
@@ -0,0 +1,97 @@
+<?php
+
+namespace Icinga\Util;
+
+use Icinga\Web\Response;
+use Icinga\Web\Session;
+use Icinga\Web\Window;
+
+class Csp
+{
+    /** @var self */
+    private static $instance = null;
+
+    /** @var string */
+    private $styleNonce = null;
+
+    /** @var string */
+    private $scriptNonce = null;
+
+    private function loadNoncesFromSession()
+    {
+        $instance = self::getInstance();
+        $windowSession = Window::getInstance()->getSessionNamespace('csp');
+
+        $sessionStyleNonce = $windowSession->get('csp_style_nonce');
+        $sessionScriptNonce = $windowSession->get('csp_script_nonce');
+
+        if (! empty($sessionStyleNonce)) {
+            $instance->styleNonce = $sessionStyleNonce;
+        }
+
+        if (! empty($sessionScriptNonce)) {
```

Comment:
We don't have use cases for this right now, right? I would only introduce this once we really need it. The separate method name for style is just a preparation.

Reply:
It's currently not being used. If the method should remain, it's going to always return `null`. Or should the method also be commented out for now?

Reply:
It's okay to remove everything related to script nonce. We can easily add it, if we need it.